### DISCUSSION TITLE
월드 플레이 중 캐릭터 이동속도 증가

### DISCUSSION THREAD

**Discussion by diddntjd99** [naverz/zepeto-studio-kor#5] [5. Q&A]
안녕하세요, 기존에 알려주셨던
const zepetoPlayer = ZepetoPlayers.instance.GetPlayer(this.room.SessionId);
zepetoPlayer.character.additionalRunSpeed += 10;
코드를 사용해도 이동속도가 안올라가지는데, 확인부탁드립니다.

**Comment by diddntjd99** [comment]
비슷한 추가적인 질문으로 캐릭터의 위치를 teleport로 변경하려 하는데,
const zepetoPlayer = ZepetoPlayers.instance.GetPlayer(this.room.SessionId);
zepetoPlayer.character.Teleport(new UnityEngine.Vector3(0, 1, 0), UnityEngine.Quaternion.Euler(0, 0, 0));

이처럼 코드를 작성해서 사용해보면 몇 번은 해당 위치로 이동하고 몇 번은 이동하지 않는데 이유를 모르겠습니다.. 해결 할 방법이 있을까요?

**Comment by moondory77-naverz** [comment]
안녕하세요. 답변이 늦어 죄송합니다.

1. 
**const zepetoPlayer = ZepetoPlayers.instance.GetPlayer(this.room.SessionId);
zepetoPlayer.character.additionalRunSpeed += 10;
코드를 사용해도 이동속도가 안올라가지는데, 확인부탁드립니다.**

=> 캐릭터 움직임에 대한 3가지 가중치
 additionalWalkSpeed / additionalRunSpeed / additionalJumpPower  값이
ZepetoCharacter 객체의 초기화 시점에만 적용되고, 이후 런타임에는 변경 시에도 적용되지 않는 이슈를 확인했습니다.
**ZEPETO.World-1.0.6** 에 수정 적용되었습니다. 업데이트 후, 확인을 부탁드리겠습니다. 
불편을 끼쳐드려 죄송합니다.

2. 
**const zepetoPlayer = ZepetoPlayers.instance.GetPlayer(this.room.SessionId);
zepetoPlayer.character.Teleport(new UnityEngine.Vector3(0, 1, 0), UnityEngine.Quaternion.Euler(0, 0, 0));
이처럼 코드를 작성해서 사용해보면 몇 번은 해당 위치로 이동하고 몇 번은 이동하지 않는데 이유를 모르겠습니다**

=> 말씀주신 내용만으로는 단서가 조금 부족한듯 합니다. 
(우선, GetPlayer를 통해 받아온 zepetoPlayer 객체에 대한 null 체크가 필요할 듯 합니다.)
혹시 계속 해당 문제가 발생할 경우, 조금 더 상황의 범위를 좁힐 수 있는 단서를 알려주시면 추가 도움을 드릴 수 있도록 하겠습니다.

감사합니다. :)
또 문의가 있으실 경우, 언제든 말씀 부탁드리겠습니다.

**Reply by diddntjd99** [reply]
@moondory77-naverz 
답변 감사합니다.
일정 타임이 되면 캐릭터의 위치를 옮기기 위해 텔레포트 함수를 사용한건데, 여러번 플레이를 해보면 이동이 될 때도 있지만 안될때도 있습니다. 어떠한 문제 때문인지 몰라서 뭔가 더 알려드릴 수 있는게 없네요..ㅠ
